test/python/test_tensorboard_ngraph_stats.py

```diff
@@ -0,0 +1,212 @@
+        y_conv, keep_prob = deepnn(x)
+
+        with tf.name_scope('loss'):
+            cross_entropy = tf.nn.softmax_cross_entropy_with_logits(
+                labels=y_, logits=y_conv)
+            cross_entropy = tf.reduce_mean(cross_entropy)
+
+        with tf.name_scope("adam_optimizer"):
+            train_step = tf.train.AdamOptimizer(1e-4).minimize(cross_entropy)
+
+        with tf.name_scope('accuracy'):
+            correct_prediction = tf.equal(
+                tf.argmax(y_conv, 1), tf.argmax(y_, 1))
+            correct_prediction = tf.cast(correct_prediction, tf.float32)
+
+            accuracy = tf.reduce_mean(correct_prediction)
+
+        tf.summary.scalar('Training accuracy', accuracy)
+        tf.summary.scalar('Loss function', cross_entropy)
+
+        graph_location = os.environ[
+            'NGRAPH_TF_TB_LOGDIR'] if 'NGRAPH_TF_TB_LOGDIR' in os.environ else "./test_tensorboard"
+
+        if (os.path.isdir(graph_location)):
+            shutil.rmtree(graph_location)
+
+        print('Saving graph to: %s' % graph_location)
+
+        merged = tf.summary.merge_all()
+        train_writer = tf.summary.FileWriter(graph_location)
+        train_writer.add_graph(tf.get_default_graph())
+
+        saver = tf.train.Saver()
+
+        with tf.Session(config=config) as sess:
+            sess.run(tf.global_variables_initializer())
+            loss_values = []
+
+            for i in range(300):
+                batch = mnist.train.next_batch(100, shuffle=False)
+                if i % 10 == 0:
+                    train_accuracy = accuracy.eval(feed_dict={
+                        x: batch[0],
+                        y_: batch[1],
+                        keep_prob: 1.0
+                    })
+
+                _, summary, loss = sess.run([train_step, merged, cross_entropy],
+                                            feed_dict={
+                                                x: batch[0],
+                                                y_: batch[1],
+                                                keep_prob: 0.5
+                                            })
+                loss_values.append(loss)
+                train_writer.add_summary(summary, i)
+
+            num_test_images = 100
+            x_test = mnist.test.images[:num_test_images]
+            y_test = mnist.test.labels[:num_test_images]
+
+            test_accuracy = accuracy.eval(feed_dict={
+                x: x_test,
+                y_: y_test,
+                keep_prob: 1.0
+            })
+
+            saver.save(sess, './mnist_trained/')
+
+        dirs = [o for o in os.listdir(graph_location)]
+
+        check = (len(dirs) == 8) and ("ngraph0_init" in dirs) and (
+            "ngraph1_accuracy" in dirs) and ("ngraph2_loss" in dirs) and (
+                "ngraph3_save" in dirs) and ("stats0_init" in dirs) and (
+                    "stats1_accuracy" in dirs) and ("stats2_loss" in dirs)
+
+        assert (check)
```

Comment:
i would assert them indivisually, so that the error message can reflect which one of the many tests actually failed.

I mean:
```
assert len(dirs) == 8
assert "ngraph0_init" in dirs
...
```